src/component/mxgraph/shape/render/icon-painter.ts

```diff
@@ -354,6 +354,23 @@ export class IconPainter {
     return canvas;
   }
 
+  /**
+   * Code for creating an asterisk icon used by `complex gateway`.
+   */
+  private createCanvas(paintParameter: PaintParameter): BpmnCanvas {
+    const canvas = this.newBpmnCanvas(paintParameter, { height: 1, width: 1 });
+    return canvas;
+  }
+
+  private drawVerticalIconOnCanvas(paintParameter: PaintParameter, canvas: BpmnCanvas): void {
```

Comment:
can you move this function out of the class as it does not rely on any internal property?

Reply:
I think it would be better to move it in `BpmnCanvas` class and rename it in `drawVerticalLine`.

Reply:
For now, I just renamed the function in 4168469c41166b10e75b3d7b376856e560abd74d

Reply:
I moved the function in 48daabf